src/sqlfluff/__init__.py

```diff
@@ -20,6 +20,9 @@
 import pkg_resources
 import configparser
 
+# Increase the maximum recursion depth from default of 10^4 to 10^5, see https://github.com/sqlfluff/sqlfluff/pull/870
+sys.setrecursionlimit(10 ** 5)
```

Comment:
TBH, I'm surprised that increasing it to such a high number:
* Is necessary
* Works

Usually, if I write code that recurses this deeply, it's because of accidental _infinite_ recursion. Do we have a sense for why the parser recurses so deeply but eventually does the right thing and stops?

Reply:
TBH, I don't know what units this number is in. I had naively assumed it is the number of function calls. Is it possible it's something else, such as bytes, e.g. the amount of storage taken up the stack, local variables, ??